crypto/dilithium/p_dilithium3.c

```diff
@@ -0,0 +1,91 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <openssl/evp.h>
+#include <openssl/err.h>
+#include <openssl/mem.h>
+
+#include "../fipsmodule/evp/internal.h"
+#include "../evp_extra/internal.h"
+#include "sig_dilithium.h"
+
+static int pkey_dilithium3_keygen(EVP_PKEY_CTX *ctx, EVP_PKEY *pkey) {
+  DILITHIUM3_KEY *key = OPENSSL_malloc(sizeof(DILITHIUM3_KEY));
+  if (key == NULL) {
+    OPENSSL_PUT_ERROR(EVP, ERR_R_MALLOC_FAILURE);
+    return 0;
+  }
+
+  if (!EVP_PKEY_set_type(pkey, EVP_PKEY_DILITHIUM3)) {
+    OPENSSL_free(key);
+    return 0;
+  }
+
+  DILITHIUM3_keypair(key->pub, key->priv);
```

Comment:
I'm trying to be consistent with [p_kyber.c](https://github.com/awslabs/aws-lc/blob/main/crypto/kyber/p_kyber.c#L22) [p_ed25519.c](https://github.com/awslabs/aws-lc/blob/main/crypto/evp_extra/p_ed25519.c#L41) and [p_x25519.c](https://github.com/awslabs/aws-lc/blob/main/crypto/evp_extra/p_x25519.c#L40) which all do it this way...

Reply:
Is that a bug in those spots as well? @dkostic thoughts?

Reply:
I don't think it's a bug, those functions always return successfully so I don't see a reason to check the return value.

Reply:
discussed this with Torben a bit -- it's probably a good idea to always check the return values in case the functions change at some point and start returning something meaningful. So let's add the checks.